util/predicates/generic_predicates.go

```diff
@@ -303,3 +308,28 @@ func processIfTopologyOwned(scheme *runtime.Scheme, logger logr.Logger, obj clie
 	logger.V(6).Info("Resource is not topology owned, will not attempt to map resource")
 	return false
 }
+
+// ResourceIsChanged returns a predicate that returns true only if the resource
+// has changed. This predicate allows to drop resync events on additionally watched objects.
+func ResourceIsChanged(logger logr.Logger) predicate.Funcs {
+	return TypedResourceIsChanged[client.Object](logger)
+}
+
+// TypedResourceIsChanged returns a predicate that returns true only if the resource
+// has changed. This predicate allows to drop resync events on additionally watched objects.
+func TypedResourceIsChanged[T client.Object](logger logr.Logger) predicate.TypedFuncs[T] {
+	log := logger.WithValues("predicate", "ResourceIsTopologyOwned")
+	return predicate.TypedFuncs[T]{
+		UpdateFunc: func(e event.TypedUpdateEvent[T]) bool {
```

Comment:
Sorry I think we also have to add the scheme to add the kind: namespace/name k/v pair.

Otherwise the logs are not really useful

Plesae make sure that when we add something like this in UpdateFunc
```go
	if gvk, err := apiutil.GVKForObject(obj, scheme); err == nil {
		logger = logger.WithValues(gvk.Kind, klog.KObj(obj))
	}
```
We don't end up with a logger that has one additional k/v pair with every invocation of the UpdateFunc

Reply:
Should have everywhere now the predicateLogger which does not have any kv pairs except the controller name added.